docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,392 @@
+    }
+
+.. code-block:: yaml
+
+    App\Checker\ResourceChannelEnabilibityChecker:
+        decorates: Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface
+        arguments: ['@.inner']
+
+* Create ``App\Checker\ResourceChannelChecker`` and decorate ``Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface`` next add condition for checking ``Supplier``.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Checker;
+
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface;
+    use Sylius\Plus\Entity\ChannelInterface;
+
+    final class ResourceChannelChecker implements ResourceChannelCheckerInterface
+    {
+        /** @var ResourceChannelCheckerInterface */
+        private $decoratedResourceChannelChecker;
+
+        public function __construct(ResourceChannelCheckerInterface $decoratedResourceChannelChecker)
+        {
+            $this->decoratedResourceChannelChecker = $decoratedResourceChannelChecker;
+        }
+
+        public function isFromChannel(object $resource, ChannelInterface $channel): bool
+        {
+                // this condition is needed because while creating a Symfony form (our edit/create section), the given object has all fields as null. After creating an empty form this iteration is repeated and properly checked.            if ($resource instanceof Supplier && $resource->getChannel() === null) {
+                return true;
+            }
+
+            return $this->decoratedResourceChannelChecker->isFromChannel($resource, $channel);
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Checker\ResourceChannelChecker:
+        decorates: Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface
+        arguments: ['@.inner']
+
+After that, access to the resource should work properly with all restrictions.
+
+1. Extend the resource repository's method used for getting a list of Suppliers
+
+* In order to view only Suppliers from the Channel to which the current administrator has access on the index of Suppliers, we need to extend the Supplier resource repository with ``getQueryBuilder()`` method:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Repository;
+
+    use Doctrine\ORM\QueryBuilder;
+    use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+
+    final class SupplierRepository extends EntityRepository
+    {
+        public function getQueryBuilder(): QueryBuilder
+        {
+            return $this->createQueryBuilder('o');
+        }
+    }
+
+* Remember to register ``App\Repository\SupplierRepository`` for resource:
+
+.. code-block:: yaml
+
+    sylius_resource:
+    resources:
+        app.supplier:
+            driver: doctrine/orm
+            classes:
+                model: App\Entity\Supplier
+                form: App\Form\SupplierType
+  +             repository: App\Repository\SupplierRepository
```

Comment:
I'm almost sure that this part is not needed as the `getQueryBuilder` method exists in `EntityRepository`. Could you test that?

Reply:
`getQueryBuilder` doesn't exist but `createQueryBuilder` yes, I tried with it but I have problems.